src/oop/special_class_method/hard/q3/MainTest.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    private static Stream<Arguments> setSizeInputProvider() {
+        return Stream.of(
+                Arguments.of(0.0, 0.0),
+                Arguments.of(22431.21, 6669.21458764),
+                Arguments.of(1, 0.0),
+                Arguments.of(843739243.2229, 163.35)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("setSizeInputProvider")
+    public void chairClassHasCorrectSetSizeMethod(double initialValue, double setValue) throws Throwable {
+        Object[][] arguments = {
+                {initialValue, double.class},
+                {"Initial String", String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, double.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setDoubleMethodName, "does not correctly set the value of the", doubleFieldName, "field.");
+        Object setMethodOutput = testClass.callMethod(setDoubleMethodName, setSizeArguments, setMethodModifiers, classInstance);
+        _assertEquals(setValue, testClass.getFieldValue(classInstance, doubleFieldName), incorrectSetMethodMessage);
+        assertNull(setMethodOutput, String.join(" ", "Your", setDoubleMethodName, "should not return any output"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void chairClassHasCorrectGetSizeMethod(double value) throws Throwable {
+        Object[][] arguments = {
+                {value, double.class},
+                {"Initial String", String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] getMethodModifiers = {"public"};
+        String incorrectGetMethodMessage = String.join(" ",
+                "Your", getDoubleMethodName, "does not correctly get the value of the", doubleFieldName, "field.");
+        Object getMethodOutput = testClass.callMethod(getDoubleMethodName, getMethodModifiers, classInstance);
+        _assertEquals(value, getMethodOutput, incorrectGetMethodMessage);
+    }
+
+    private static Stream<Arguments> twoStringInputProvider() {
+        return Stream.of(
+                Arguments.of("Short", "Tall"),
+                Arguments.of("Window", "Cup"),
+                Arguments.of("Blue", "Red"),
+                Arguments.of("Horizontal", "Vertical")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoStringInputProvider")
+    public void chairClassHasCorrectSetSoftnessMethod(String initialValue, String updatedValue) throws Throwable {
+        Object[][] arguments = {
+                {Math.PI, double.class},
+                {initialValue, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        testClass.setFieldValue(classInstance, initialValue, StringFieldName);
```

Comment:
why is this here? Doesn't creating the instance with the arguments already do that?